go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -751,6 +752,24 @@ func (tm *TabletManager) findMysqlPort(retryInterval time.Duration) {
 	}
 }
 
+// redoPreparedTransactionsAndSetReadWrite redoes prepared transactions in read-only mode.
+// We turn off super read only mode, and then redo the transactions. Finally, we turn of read-only mode to allow for further traffic.
+func (tm *TabletManager) redoPreparedTransactionsAndSetReadWrite(ctx context.Context) error {
+	_, err := tm.MysqlDaemon.SetSuperReadOnly(ctx, false)
+	if err != nil {
+		// Ignore the error if the sever doesn't support super read only variable.
+		// We should just redo the preapred transactions before we set it to read-write.
+		if sqlErr, ok := err.(*sqlerror.SQLError); ok && sqlErr.Number() == sqlerror.ERUnknownSystemVariable {
+			log.Warningf("server does not know about super_read_only, continuing anyway...")
```

Comment:
why do we expect this error?

Reply:
`InitPrimary` can be called on a tablet running mariadb. It is required for that.

Reply:
AFAIK, Mariadb 10.1 onwards know about super_read_only variable

Reply:
It is fine by me to keep this check

Reply:
I thought support for Maria was dropped in v14 in any case?